javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -50,6 +50,10 @@ predicate benignContext(Expr e) {
   exists(VoidExpr voidExpr | voidExpr.getOperand() = e)
 
   or
+  // weeds out calls inside HTML-attributes.
+  e.getContainer() instanceof CodeInAttribute or  
```

Comment:
This is extremely unlikely to matter in practice, but asking for the container means that `g()` in `<a onclick='alert("msg: " + g())'/>` will not be flagged. 

See:

https://github.com/Semmle/ql/blob/39f550b6d2bea3deaaec1585973ce1c5eed687ce/javascript/ql/src/Expressions/ExprHasNoEffect.qll#L13-L16

Reply:
I initially had the `getParent*() instanceof CodeInAttribute` just to be conservative, and that is why I changed it to `getContainer()`. 

But yeah, I'll do `e.getParent() instanceof CodeInAttribute`.